[ci-visibility] Add Instructions to set custom tags and metrics (#14866)
DataDog/documentation: content/en/continuous_integration/setup_tests/javascript.md

NODE_OPTIONS="-r $(pwd)/.pnp.cjs -r dd-trace/ci/init" yarn test
{{< /code-block >}}

> ### Adding custom tags and metrics to testss

Reviewer: ```suggestion
### Adding custom tags and metrics to tests
```

> You can add custom tags to your test by grabbing the current active span:

Reviewer 2: ```suggestion
You can add custom tags to your tests by using the current active span:
```

```javascript
  it('sum function can sum', () => {
    const testSpan = require('dd-trace').scope().active()
    testSpan.setTag('team.owner', 'calculator')
    testSpan.setTag('test.importance', 2)
    // test continues normally
    // ...
  })
```

> To create filters or group bys for these tags, you need to create facets for them. For more information about custom instrumentation read the [NodeJS Custom Instrumentation][1] documentation.

Reviewer 3: ```suggestion
To create filters or `group by` fields for these tags, you must first create facets. For more information about custom instrumentation, see the [NodeJS Custom Instrumentation documentation][1].
```

> If you want to create filters or group bys for these tags you need to create facets for them. For more information about custom instrumentation you can read [NodeJS Custom Instrumentation][1].

Reviewer 2: ```suggestion
To create filters or group bys for these tags, you need to create facets for them. For more information about custom instrumentation read the [NodeJS Custom Instrumentation][1] documentation.
```

[1]: /tracing/trace_collection/custom_instrumentation/nodejs?tab=locally#adding-tags
{{% /tab %}}

{{% tab "Cucumber" %}}

> Set `NODE_OPTIONS` environment variable to `-r dd-trace/ci/init`. Run your tests as you normally would, specifying the environment where the tests are run in the `DD_ENV` environment variable. For example, set `DD_ENV` to `local` when running tests on a developer workstation, or `ci` when running them on a CI provider:

Reviewer 2: ```suggestion
Set the `NODE_OPTIONS` environment variable to `-r dd-trace/ci/init`. Run your tests as you normally would, specifying the environment where the tests are run in the `DD_ENV` environment variable. For example, set `DD_ENV` to `local` when running tests on a developer workstation, or `ci` when running them on a CI provider:
```

{{< code-block lang="bash" >}}
NODE_OPTIONS="-r dd-trace/ci/init" DD_ENV=ci DD_SERVICE=my-javascript-app yarn test
{{< /code-block >}}

### Using Yarn >=2

If you're using `yarn>=2` and a `.pnp.cjs` file, and you get the following error message when using `NODE_OPTIONS`:

```text
 Error: Cannot find module 'dd-trace/ci/init'
```

> You can fix it by setting `NODE_OPTIONS` to the following:

Reviewer 3: ```suggestion
You can fix this by setting `NODE_OPTIONS` to the following:
```

{{< code-block lang="bash" >}}
NODE_OPTIONS="-r $(pwd)/.pnp.cjs -r dd-trace/ci/init" yarn test
{{< /code-block >}}

### Adding custom tags to tests

You can add custom tags to your test by grabbing the current active span:

```javascript
  When('the function is called', function () {
    const stepSpan = require('dd-trace').scope().active()
    stepSpan.setTag('team.owner', 'calculator')
    stepSpan.setTag('test.importance', 2)
    // test continues normally
    // ...
  });
```

> To create filters or group bys for these tags, you must create facets for them. For more information about custom instrumentation, read the [NodeJS Custom Instrumentation][1] documentation.

Reviewer 3: ```suggestion
To create filters or `group by` fields for these tags, you must first create facets. For more information about custom instrumentation, see the [NodeJS Custom Instrumentation documentation][1].
```

> If you want to create filters or group bys for these tags you need to create facets for them. For more information about custom instrumentation you can read [NodeJS Custom Instrumentation][1].

Reviewer 2: ```suggestion
To create filters or group bys for these tags, you must create facets for them. For more information about custom instrumentation, read the [NodeJS Custom Instrumentation][1] documentation.
```

{{< /code-block >}}

### Adding custom tags to tests

To add additional information to your tests, such as the team owner, use `cy.task('dd:addTags', { yourTags: 'here' })` in your test or hooks.

For example:

```javascript
beforeEach(() => {
  cy.task('dd:addTags', {
    'before.each':
    'certain.information'
  })
})
it('renders a hello world', () => {
  cy.task('dd:addTags', {
    'team.owner': 'ui',
    'test.importance': 3
  })
  cy.get('.hello-world')
    .should('have.text', 'Hello World')
})
```

> To create filters or group bys for these tags, you must create facets for them. For more information about custom instrumentation, read the [NodeJS Custom Instrumentation][1] documentation.

Reviewer 3: ```suggestion
To create filters or `group by` fields for these tags, you must first create facets. For more information about custom instrumentation, see the [NodeJS Custom Instrumentation documentation][1].
```

> If you want to create filters or group bys for these tags you need to create facets for them. For more information about custom instrumentation you can read [NodeJS Custom Instrumentation][1].

Reviewer 2: ```suggestion
To create filters or group bys for these tags, you must create facets for them. For more information about custom instrumentation, read the [NodeJS Custom Instrumentation][1] documentation.
```
